tests/run/mixin-primitive-on-generic-1.scala

```diff
@@ -0,0 +1,19 @@
+
+object Test {
+  def main(args: Array[String]): Unit = {
+    println((new Foo: Baz).value1)
+    println((new Foo: Baz).value2)
+  }
+}
+
+class Foo extends Bar[Boolean](true) with Baz
```

Comment:
need a test with a value class, also it would be nice to check that the method called would be the unboxed one.